### PR TITLE
351 - Input component work

### PR DESCRIPTION
**Pull request by Oreganop**

### CODE REVIEW

File: site/app/form.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit, Input } from '@angular/core';
```
Comment (tabersays) [inline site/app/form.ts]: You shouldn't add files to the site directory. The majority of your work should be in src.

File: site/app/app.module.ts [hunk 2]
```diff
@@ -1,21 +1,18 @@
-import { NgModule } from '@angular/core';
```
Comment (tabersays) [inline site/app/app.module.ts]: This is our entry point, the ounly thing that you should ever do in this part of the app is add you component.

File: site/app/app.component.ts [hunk 3]
```diff
@@ -1,24 +1,45 @@
-import { Component } from '@angular/core';
```
Comment (tabersays) [inline site/app/app.component.ts]: The only thing you should ever have to do is add you component and set up its config maybe. You definitely should not be removing anything.

File: site/app/input.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Component, Input } from '@angular/core';
```
Comment (tabersays) [inline site/app/input.ts]: This should also be in src/components

File: src/components/textinput.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { Component } from '@angular/core';
```
Comment (glawson) [inline src/components/textinput.ts]: rename to form-text-input

File: src/components/textinput.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+declare const module: any;
+
+@Component({
+    moduleId: module.id,
+    selector: 'bossy-textinput',
```
Comment (glawson) [inline src/components/textinput.ts]: bossy-text-input

File: src/components/textinput.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    templateUrl: '../templates/textinput.html',
```
Comment (glawson) [inline src/components/textinput.ts]: rename to pattern

File: src/components/textinput.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    styleUrls: ['../styles/textinput.css'],
```
Comment (glawson) [inline src/components/textinput.ts]: same as above

File: src/components/textinput.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    inputs: [
+        'config'
+    ]
+})
+export class Bossytextinput {
```
Comment (glawson) [inline src/components/textinput.ts]: BossyFormTextInput

File: src/components/textinput.ts (hunk 7 above)
Comment (glawson) [inline src/components/textinput.ts]: templateUrl should be '../templates/form-input.html

**Pull request merged**